### ISSUE TITLE
Patching ASM xml silently fails

### ISSUE THREAD

**Issue by invadermonks** [Glain/FFTPatcher#22]
When patching a large ASM XML (created in the gui by selecting the ASMs and clicking Patch->Create XML) FFTorgASM will not make any changes to the disc image and will still report success. When patching a simple one entry ASM XML it will succeed. Command output (with MD5 hashing to show when the disc image does or does not change) is pasted below and I am attaching the XML files used.

[asm_xmls.zip](https://github.com/Glain/FFTPatcher/files/11588798/asm_xmls.zip)

```
C:\Users\invadermonks\Documents\fft_test>copy fft\fft.bin output\fft_1685338731.bin
        1 file(s) copied.

C:\Users\invadermonks\Documents\fft_test>certutil -hashfile output\fft_1685338731.bin MD5
MD5 hash of output\fft_1685338731.bin:
b156ba386436d20fd5ed8d37bab6b624
CertUtil: -hashfile command completed successfully.

C:\Users\invadermonks\Documents\fft_test>start /wait "Patch" tools\fftpatcher\FFTorgASM.exe  -patch data\tlw_asm.xml output\fft_1685338731.bin
(Success) Complete!



C:\Users\invadermonks\Documents\fft_test>certutil -hashfile output\fft_1685338731.bin MD5
MD5 hash of output\fft_1685338731.bin:
b156ba386436d20fd5ed8d37bab6b624
CertUtil: -hashfile command completed successfully.


C:\Users\invadermonks\Documents\fft_test>start /wait "Patch" tools\fftpatcher\FFTorgASM.exe  -patch data\asm\TLW_Serial.xml output\fft_1685338731.bin
(Success) Complete!



C:\Users\invadermonks\Documents\fft_test>certutil -hashfile output\fft_1685338731.bin MD5

MD5 hash of output\fft_1685338731.bin:
e07bfa850457157ef3164ddc37877a1d
CertUtil: -hashfile command completed successfully.
```

**Comment by invadermonks** [comment]
It seems like the problem is either in the "Create XML" functionality or in loading XML files over a certain size. Trying to load the 3MB tlw_asm.xml gives the following error in the gui: "Invalid sector name!"

Patching the folder of individual XML files that I used to create the merged xml files works fine both in the GUI and the cmdline (giving the same resultant MD5).

**Comment by Glain** [comment]
There are a few facets to this, but they should all be fixed with these commits: 2fbbd29, 31c9619, 5636425.

**Comment by invadermonks** [comment]
I did the same test on the current code and I was able to generate an xml of all of the TLW asms and patching it gives the same md5 as patching each asm xml individually now. Thanks!
